sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/Utility.java

```diff
@@ -1348,4 +1349,37 @@ public static SentenceAssessment findSentimentAssessment(String assessmentPointe
         }
         return assessments.get(assessmentIndex);
     }
+
+    /**
+     * Throw exception if sourceVersion is one of target API versions.
+     *
+     * @param sourceVersion The service version that client is using when calling service API.
+     * @param targetVersions The target versions that used to verify if sourceVersion is unsupported version.
+     *                       Throw exception if target versions contains the source version.
+     * @param errorMessage The error message.
+     */
+    public static void throwIfTargetServiceVersionFound(TextAnalyticsServiceVersion sourceVersion,
+                                                        List<TextAnalyticsServiceVersion> targetVersions,
+                                                        String errorMessage) {
+        for (TextAnalyticsServiceVersion targetVersion : targetVersions) {
+            if (targetVersion != null && sourceVersion != null
+                && targetVersion.getVersion().equals(sourceVersion.getVersion())) {
+                throw LOGGER.logExceptionAsError(new IllegalStateException(errorMessage));
+            }
+        }
+    }
+
+    /**
+     * Retrieve custom unsupported Service API version error message.
+     *
+     * @param unsupportedName The unsupported API or property name that the not available in 'minSupportedVersion'.
+     * @param sourceVersion The source service API version that client is using.
+     * @param minSupportedVersion The minimum supported Service API version.
+     * @return The error message.
+     */
+    public static String getUnsupportedServiceApiVersionMessage(String unsupportedName,
+        TextAnalyticsServiceVersion sourceVersion, TextAnalyticsServiceVersion minSupportedVersion) {
+        return String.format("'%s' is not available in API version %s. Use service API version '%s' or newer.",
```

Comment:
Is it clear to the customer that v2022-05-01 is newer than v3.1? Should the message be clearer in that regard?

Reply:
I think we are suggesting using the min version of 2022-05-01. It is inclusive. Customers will find out which version is newer from CHANGELOG history, README, and the order of TextAnalyticsServiceVersion.

Reply:
> I think we are suggesting using the min version of 2022-05-01

If the customer is trying to use `disableServiceLogs` with v3.0, would the error message suggest using v3.1 and newer or v2022-05-01 and newer?

Reply:
for `disableServiceLogs`,  we would suggest the customer to use v3.1 at least. 
https://github.com/Azure/azure-sdk-for-java/pull/30708/files#diff-e6b45745e42ad20893ffc77376d77776b9e455cfa18486a3d6eee01676072804R186